odoo/models.py

```diff
@@ -4675,13 +4675,13 @@ def exists(self):
 
         By convention, new records are returned as existing.
         """
-        ids, new_ids = [], []
-        for i in self._ids:
-            (new_ids if isinstance(i, NewId) else ids).append(i)
+        new_ids, ids = partition(lambda i: isinstance(i, NewId), self._ids)
```

Comment:
As `new_ids` contains the falsy ones, you could write this instead:
```suggestion
        ids, new_ids = partition(bool, self._ids)
```

Reply:
I understand the idea, but it seems dangerous if the id == 0 (or empty string, or other Falsy) (generate by a custom SQL views or table query) it will considered as a NewId even if it isn't. Well It is a very specific corner case but we never know.
Here a example:
https://github.com/odoo/odoo/blob/14.0/addons/website_crm_partner_assign/report/crm_partner_report.py#L40 
for the 1 000 000 000 partner id, it will always (well this code is weird anyway, but it is in stable).
Or we don't care about these cases (that shouldn't be exist anyway) you prefer using bool ?

Reply:
Alright, thanks. I will change the rapport too then, and use bool.

Reply:
Mhh one ORM test fail due to the `bool` change: " AssertionError: res.partner(0,) is not false "
Should I change it ?

Reply:
Let's keep it this way, then.